proto/block.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+import "digest.proto";
+
+message BlockHeader {
+    // The hash of the previous block.
+    Digest prev_hash = 1;
+
+    // This block's number/height.
+    uint64 block_num = 2;
+
+    // A commitment to the chain's MMR.
+    Digest chain_root = 3;
+
+    // A commitment to the account's and nullifier's databases.
+    Digest state_root = 4;
+
+    // A commitment to the note's created in this block.
+    Digest note_root = 5;
+
+    // A commitment to the transaction batch executed in this block.
+    Digest batch_root = 6;
+    
+    // The hash of the stark proof attesting the block.
+    Digest proof_hash = 7;
+
+    // The result of `hash(prev_hash, chain_root, state_root, batch_root, proof_hash, block_num)`.
+    Digest sub_hash = 8;
+
+    // The result of `hash(sub_hash, note_root)`.
+    Digest hash = 9;
```

Comment:
Should these be a part of the message? My thinking was that values like these would be computed based on the other data.

Reply:
Makes sense, changing.